modules/shared/src/main/scala/org/tessellation/schema/trust.scala

```diff
@@ -95,22 +96,32 @@ object trust {
   @derive(decoder, encoder, show, eqv)
   case class PeerObservationAdjustmentUpdateBatch(updates: List[PeerObservationAdjustmentUpdate])
 
-  @derive(decoder, encoder, show)
+  @derive(decoder, eqv, encoder, show)
   case class TrustInfo(
     trustLabel: Option[Double] = None,
     predictedTrust: Option[Double] = None,
     observationAdjustmentTrust: Option[Double] = None,
     peerLabels: Map[PeerId, Double] = Map.empty
   ) {
 
-    val publicTrust: Option[Double] =
+    lazy val publicTrust: Option[Double] =
       trustLabel
         .map(t => Math.max(-1, t + observationAdjustmentTrust.getOrElse(0d)))
         .orElse(observationAdjustmentTrust.map(t => Math.max(-1, t)))
   }
 
-  @derive(decoder, encoder, show)
+  @derive(eqv, decoder, encoder, show)
   case class PublicTrust(
     labels: Map[PeerId, Double]
+  ) {
+
+    def isNonEmpty: Boolean = labels =!= Map.empty
```

Comment:
you can see that scala uses `isEmpty` and `nonEmpty` methods (like on Set, List, ...) so I think it would be good to use the same naming. wdyt?

Reply:
Changed to

```
def isEmpty: Boolean = labels === Map.empty
```